### ISSUE TITLE
GPU profiling: cuptiGetTimestamp: error 999 

### ISSUE THREAD

**Issue by silvasean** [google/jax#16030]
### Description

I have been trying to profile a program running on my A100 GPU on Google Cloud.

When I wrap my computation with
```
with jax.profiler.trace("/tmp/jax-trace", create_perfetto_link=True):
```
as described in the docs I get out a perfetto link, but it doesn't have any GPU information. Looking in the logs, I see lines like this:

```
2023-05-16 21:01:05.541307: E external/xla/xla/backends/profiler/gpu/cupti_error_manager.cc:137] cuptiGetTimestamp: error 999: 
2023-05-16 21:01:05.541339: E external/xla/xla/backends/profiler/gpu/cupti_error_manager.cc:186] cuptiSubscribe: ignored due to a previous error.
2023-05-16 21:01:05.541345: E external/xla/xla/backends/profiler/gpu/cupti_error_manager.cc:459] cuptiGetResultString: ignored due to a previous error.
2023-05-16 21:01:05.541350: E external/xla/xla/backends/profiler/gpu/cupti_tracer.cc:1722] function cupti_interface_->Subscribe( &subscriber_, (CUpti_CallbackFunc)ApiCallback, this)failed with error 
```

Any idea how to further debug or fix this?

(I tested the workaround in https://github.com/google/jax/issues/13477 but it did not fix things)

### What jax/jaxlib version are you using?

jax=0.4.8, jaxlib=0.4.7+cuda11.cudnn86

### Which accelerator(s) are you using?

GPU (A100)

### Additional system info

22.04 Ubuntu

### NVIDIA GPU info

https://gist.github.com/silvasean/2f6da6081642c59b9e77711f3a472fc1

**Comment by hawkinsp** [comment]
Did you install CUDA using the `pip` packages? If so, can you try updating to the latest release (0.4.10)?

I think this may have already been fixed by https://github.com/google/jax/commit/75d0f6522d3e4623e7fd5bd512645e4cbab6b10d
which added a missing `cupti` dependency.

**Comment by silvasean** [comment]
Thanks Peter. I see `nvidia-cuda-cupti-cu11` already installed in my venv:

```
$ python -m pip list | grep nvidia
nvidia-cublas-cu11                11.11.3.6
nvidia-cuda-cupti-cu11            11.8.87
nvidia-cuda-nvcc-cu11             11.8.89
nvidia-cuda-runtime-cu11          11.8.89
nvidia-cudnn-cu11                 8.9.0.131
nvidia-cufft-cu11                 10.9.0.58
nvidia-cusolver-cu11              11.4.1.48
nvidia-cusparse-cu11              11.7.5.86
```

However, I also have CUDA installed locally. I suspect the local one is being picked, since `LD_LIBRARY_PATH=/usr/local/cuda-11.7/extras/CUPTI/lib64/` appears to fix the issue.

Is there a way to observe or control which CUDA installation is being picked?

**Comment by silvasean** [comment]
No, I did not update, since I found a workaround (and I find these things to be fragile as they get updated -- it sounds like the packaging situation is getting a lot more robust though so I'm excited to see that change!).

When you say `RPATH` -- which specific ELF file is most relevant here?
I looked in xla_extension.so but see no RPATH, only RUNPATH (very similar, except the order in which they are searched). (and indeed, the cupti path is missing)
```
 0x000000000000001d (RUNPATH)            Library runpath: [$ORIGIN/../nvidia/cuda_runtime/lib:$ORIGIN/../nvidia/cublas/lib:$ORIGIN/../nvidia/cufft/lib:$ORIGIN/../nvidia/cudnn/lib:$ORIGIN/../nvidia/cusolver/lib]
```

Overall, I think I have a solution for me locally here, and it looks like there is work in progress/landed to make this not an issue going forward, so feel free to close!
